src/main/java/datacollection/DataCheckThread.java

```diff
@@ -63,32 +64,38 @@ private void requestActivityData(TokenMap tokenMap) {
         try {
             // Refresh token on start
             final OAuth2AccessToken accessToken = this.fitbitClient.refreshAccessToken(tokenMap.getRefreshToken());
+            tokenMap.setAccessToken(accessToken.getAccessToken());
+            tokenMap.setRefreshToken(accessToken.getRefreshToken());
+            tokenMap.setExpiresIn(accessToken.getExpiresIn());
```

Comment:
You have updated the token map but I don't see it being committed to the DB. Is it done elsewhere?